File: scratch/check_natureza.py
import fdb

con = fdb.connect(dsn='C:\\Coliseu\\Data\\PIVETA.FDB', user='sysdba', password='masterkey')
cur = con.cursor()
cur.execute("SELECT ID_NATUREZA, DESCRICAO, OPERACAO, TIPO, PROCESSO FROM NATUREZA_OPERACAO WHERE ID_NATUREZA = 1")
row = cur.fetchone()
if row:
    print(f"ID: {row[0]}, DESCRICAO: {row[1]}, OPERACAO: {row[2]}, TIPO: {row[3]}, PROCESSO: {row[4]}")
else:
    print("Natureza 1 not found")
con.close()

File: middleware/check_natureza.js
'use strict';
const Firebird = require('node-firebird');

const options = {
    host: '127.0.0.1',
    port: 3050,
    database: 'C:\\Coliseu\\Data\\PIVETA.FDB',
    user: 'SYSDBA',
    password: 'masterkey',
    lowercase_keys: false,
};

Firebird.attach(options, (err, db) => {
    if(err) return console.error(err);
    
    const mobId = 529699;
    const erpId = 529674;
    
    const query = `
        SELECT 
            p.ID_PEDIDO, 
            p.PEDIDO, 
            p.ID_NATUREZA, 
            p.STATUS, 
            p.TIPO,
            n.OPERACAO,
            n.TIPO AS NAT_TIPO,
            n.PROCESSO AS NAT_PROCESSO,
            n.CALC_COMISSAO
        FROM PEDIDOS p
        LEFT JOIN NATUREZA_OPERACAO n ON n.ID_NATUREZA = p.ID_NATUREZA
        WHERE p.ID_PEDIDO IN (?, ?)
    `;
    
    db.query(query, [mobId, erpId], (e, r) => {
        if(e) console.error(e);
        console.table(r);
        db.detach();
    });
});
